grakn-graql/src/test/java/ai/grakn/graql/internal/gremlin/GraqlTraversalTest.java

```diff
@@ -238,55 +239,55 @@ public void makeSureTypeIsCheckedBeforeFollowingAShortcut() {
 
     @Test
-    public void whenPlanningSimpleBinaryRelationQuery_ApplyShortcutOptimisation() {
+    public void whenPlanningSimpleBinaryRelationQuery_ApplyRolePlayerOptimisation() {
         VarPattern rel = var("x").rel("y").rel("z");
 
         GraqlTraversal graqlTraversal = semiOptimal(rel);
 
         assertThat(graqlTraversal, anyOf(
-                matches("\\{\\$x-\\[shortcut:\\$.*]->\\$.* \\$x-\\[shortcut:\\$.*]->\\$.* \\$.*\\[neq:\\$.*]}"),
-                matches("\\{\\$.*<-\\[shortcut:\\$.*]-\\$x-\\[shortcut:\\$.*]->\\$.* \\$.*\\[neq:\\$.*]}")
+                matches("\\{\\$x-\\[" + Schema.EdgeLabel.ROLE_PLAYER.getLabel() + ":\\$.*]->\\$.* \\$x-\\[" + Schema.EdgeLabel.ROLE_PLAYER.getLabel() + ":\\$.*]->\\$.* \\$.*\\[neq:\\$.*]}"),
+                matches("\\{\\$.*<-\\[" + Schema.EdgeLabel.ROLE_PLAYER.getLabel() + ":\\$.*]-\\$x-\\[" + Schema.EdgeLabel.ROLE_PLAYER.getLabel() + ":\\$.*]->\\$.* \\$.*\\[neq:\\$.*]}")
         ));
     }
 
     @Test
-    public void whenPlanningBinaryRelationQueryWithType_ApplyShortcutOptimisation() {
+    public void whenPlanningBinaryRelationQueryWithType_ApplyRolePlayerOptimisation() {
         VarPattern rel = var("x").rel("y").rel("z").isa("marriage");
 
         GraqlTraversal graqlTraversal = semiOptimal(rel);
 
         assertThat(graqlTraversal, anyOf(
-                matches(".*\\$x-\\[shortcut:\\$.* rels:marriage]->\\$.* \\$x-\\[shortcut:\\$.* rels:marriage]->\\$.* \\$.*\\[neq:\\$.*].*"),
-                matches(".*\\$.*<-\\[shortcut:\\$.* rels:marriage]-\\$x-\\[shortcut:\\$.* rels:marriage]->\\$.* \\$.*\\[neq:\\$.*].*")
+                matches(".*\\$x-\\[" + Schema.EdgeLabel.ROLE_PLAYER.getLabel() + ":\\$.* rels:marriage]->\\$.* \\$x-\\[" + Schema.EdgeLabel.ROLE_PLAYER.getLabel() + ":\\$.* rels:marriage]->\\$.* \\$.*\\[neq:\\$.*].*"),
+                matches(".*\\$.*<-\\[" + Schema.EdgeLabel.ROLE_PLAYER.getLabel() + ":\\$.* rels:marriage]-\\$x-\\[" + Schema.EdgeLabel.ROLE_PLAYER.getLabel() + ":\\$.* rels:marriage]->\\$.* \\$.*\\[neq:\\$.*].*")
         ));
     }
 
     @Test
-    public void testShortcutOptimisationWithRoles() {
+    public void testRolePlayerOptimisationWithRoles() {
         VarPattern rel = var("x").rel("y").rel("wife", "z");
 
         GraqlTraversal graqlTraversal = semiOptimal(rel);
 
         assertThat(graqlTraversal, anyOf(
-                matches(".*\\$x-\\[shortcut:\\$.* roles:wife]->\\$.* \\$x-\\[shortcut:\\$.*]->\\$.* \\$.*\\[neq:\\$.*].*"),
-                matches(".*\\$.*<-\\[shortcut:\\$.* roles:wife]-\\$x-\\[shortcut:\\$.*]->\\$.* \\$.*\\[neq:\\$.*].*")
+                matches(".*\\$x-\\[" + Schema.EdgeLabel.ROLE_PLAYER.getLabel() + ":\\$.* roles:wife]->\\$.* \\$x-\\[" + Schema.EdgeLabel.ROLE_PLAYER.getLabel() + ":\\$.*]->\\$.* \\$.*\\[neq:\\$.*].*"),
+                matches(".*\\$.*<-\\[" + Schema.EdgeLabel.ROLE_PLAYER.getLabel() + ":\\$.* roles:wife]-\\$x-\\[" + Schema.EdgeLabel.ROLE_PLAYER.getLabel() + ":\\$.*]->\\$.* \\$.*\\[neq:\\$.*].*")
```

Comment:
These horrid regexes are my fault. I think it might be a good idea to make a constant `ROLE_PLAYER = Schema.EdgeLabel.ROLE_PLAYER.getLabel()` for this test, just so this doesn't get any more unreadable :(